File: 1784. Check if Binary String Has at Most One Segment of Ones/Solution.py
class Solution:
    def checkOnesSegment_MK1(self, s: str) -> bool:
        return '01' not in s

    def checkOnesSegment_MK2(self, s: str) -> bool:
        return sum(a != b for a, b in zip(s, s[1:])) < 2

File: 1784. Check if Binary String Has at Most One Segment of Ones/Solution.cpp
#include <string>
using namespace std;

class Solution
{
public:
    bool checkOnesSegment_MK1(string s)
    {
        int cnt = 0;
        for (size_t i = 1; i < s.size(); i++)
            if (s[i] ^ s[i - 1] && ++cnt > 1)
                return false;
        return true;
    }

    bool checkOnesSegment_MK2(string s)
    {
        return s.find("01") == string::npos;
    }
};
